roles/kafka_broker/tasks/health_check.yml

```diff
@@ -13,7 +13,7 @@
   ignore_errors: true
   changed_when: false
   check_mode: false
-  when: not rbac_enabled|bool
+  when: not kafka_broker_client_secrets_protection_enabled or not (rbac_enabled|bool or confluent_cli_version is version('3.0.0', '>='))
```

Comment:
can you please explain the logic behind this condition? why is there an 'or' and not 'and'?